components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -118,39 +124,65 @@ public void createBusinessRuleFromScratch(BusinessRuleFromScratch businessRuleFr
             try {
                 isDeployed = deployBusinessRule(businessRuleFromScratch);
                 saveBusinessRuleDefinition(businessRuleUUID, businessRuleFromScratch, isDeployed);
-                deployTemplates(businessRuleFromScratch);
             } catch (TemplateManagerException e) {
                 // Saving definition is unsuccessful
-                log.error("Error in saving the Business Rule definition", e); // Exception is thrown from the saveBusinessRuleDefinition method itself            }
+                log.error("Error in saving the Business Rule definition", e);
+                // Exception is thrown from the saveBusinessRuleDefinition method itself            }
             }
         } catch (Exception e) {
             log.error(e.getMessage(), e);
         }
     }
 
-
-    public void editBusinessRuleFromTemplate(String uuid, BusinessRuleFromTemplate businessRuleFromTemplate) { // todo: verify next lower level
+    public void editBusinessRuleFromTemplate(String uuid, BusinessRuleFromTemplate businessRuleFromTemplate) {
+        // todo: verify next lower level
         Map<String, Artifact> derivedTemplates = null;
+        boolean isDeployed=false;
         try {
             derivedTemplates = deriveTemplates(businessRuleFromTemplate);
         } catch (TemplateManagerException e) {
             log.error(e.getMessage(), e);
         }
 
         try {
-            overwriteBusinessRuleDefinition(uuid, businessRuleFromTemplate);
             // Load all available Business Rules again
             this.availableBusinessRules = loadBusinessRules();
             // Update Deploy templates, only if overwriting Business Rule Definition is successful
             // todo: (Q) is this ok?
             for (String templateUUID : derivedTemplates.keySet()) {
```

Comment:
Please validate whether we need to continue this loop for rest of the key, if there is exception in one key due to any of the issues( 'TemplateManagerException | UnsupportedEncodingException | SQLException | BusinessRulesDatasourceException )